src/Packetery/Module/Options/FeatureFlagManager.php

```diff
@@ -113,52 +135,41 @@ private function fetchFlags(): array {
 	 * @throws Exception From DateTimeImmutable.
 	 */
 	private function getFlags(): array {
-		static $flags;
-
-		if ( ! isset( $flags ) ) {
-			$flags = get_option( self::FLAGS_OPTION_ID );
+		if ( ! isset( self::$flags ) || null === self::$flags ) {
+			self::$flags = $this->wpAdapter->getOption( self::FLAGS_OPTION_ID );
 		}
 
-		if ( true === get_option( self::DISABLED_DUE_ERRORS_OPTION_ID ) ) {
-			return $flags ? $flags : [];
+		if ( true === $this->wpAdapter->getOption( self::DISABLED_DUE_ERRORS_OPTION_ID ) ) {
+			return self::$flags ? self::$flags : [];
 		}
 
 		$hasApiKey = ( null !== $this->optionsProvider->get_api_key() );
-		if ( false === $flags ) {
+		if ( false === self::$flags ) {
 			if ( ! $hasApiKey ) {
-				$flags = [];
+				self::$flags = [];
 
-				return $flags;
+				return self::$flags;
 			}
 
-			$flags = $this->fetchFlags();
+			self::$flags = $this->fetchFlags();
 
-			return $flags;
+			return self::$flags;
 		}
 
-		if ( $hasApiKey && isset( $flags[ self::FLAG_LAST_DOWNLOAD ] ) ) {
+		if ( $hasApiKey && isset( self::$flags[ self::FLAG_LAST_DOWNLOAD ] ) ) {
 			$now        = new DateTimeImmutable( 'now', new \DateTimeZone( 'UTC' ) );
 			$lastUpdate = DateTimeImmutable::createFromFormat(
 				Core\Helper::MYSQL_DATETIME_FORMAT,
-				$flags[ self::FLAG_LAST_DOWNLOAD ],
+				self::$flags[ self::FLAG_LAST_DOWNLOAD ],
 				new \DateTimeZone( 'UTC' )
 			);
 			$ageHours   = ( ( $now->getTimestamp() - $lastUpdate->getTimestamp() ) / HOUR_IN_SECONDS );
```

Comment:
better to use class::const instead of const.
then you can remove 

```
if ( ! defined( 'HOUR_IN_SECONDS' ) ) {
	define( 'HOUR_IN_SECONDS', 3600 );
}
```

Reply:
This constant is included in WooCommerce, I don't want to have copy in our code.